dev/InfoBadge/InfoBadge.cpp

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "InfoBadge.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+static constexpr wstring_view c_IconPresenterName{ L"IconPresenter"sv };
+
+InfoBadge::InfoBadge()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_InfoBadge);
+
+    SetDefaultStyleKey(this);
+
+    SetValue(s_TemplateSettingsProperty, winrt::make<::InfoBadgeTemplateSettings>());
+    SizeChanged({ this, &InfoBadge::OnSizeChanged });
+}
+
+void InfoBadge::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected{ *this };
+
+    OnDisplayKindPropertiesChanged();
+}
+
+winrt::Size InfoBadge::MeasureOverride(winrt::Size const& availableSize)
+{
+    auto const defaultDesiredSize = __super::MeasureOverride(availableSize);
+    if (defaultDesiredSize.Width < defaultDesiredSize.Height)
```

Comment:
Why is this needed ? Are you trying to make the layout square ? What if it is wider than taller ?

Reply:
Badge is a fully rounded rectangle that determines the rounding based on the height.  If the Width of the badge is equal to the Height then this results in a circle, if it it is larger then it is a pill shape, If the width is smaller than the height then this rounding results in a strange visualization that we don't support, hence forcing the width to be at least the height.

Reply:
can you please add a comment ? Thanks.